### DISCUSSION TITLE
Unable to create a working tree-sitter injection

### DISCUSSION THREAD

**Discussion by Mango0x45** [nvim-treesitter/nvim-treesitter#7060] [Q&A]
I write a lot of shell scripts, and the scripts often include calling to other languages (such as Awk), so it would be nice if I was able to get syntax highlighting for these other languages in string literals:

```sh
#!/bin/sh

do stuff | awk '
/pattern/ { action }  # This string should be highlighted!
/pattern/ { action }
'
```

For a good while I have solved this by using a tree-sitter injection as follows:

(~/.config/nvim/after/queries/sh/injections.scm)

```scm
;; extends

(command
  name: (command_name) @injection.language
  argument: (raw_string) @injection.content)

((command
   name: (command_name) @_name
   argument: (raw_string) @injection.content)
 (#eq? @_name "gawk")
 (#set! injection.language "awk"))
```

This worked for a good while, but for whatever reason it stopped working a good while back. I never bothered to look into it because I was busy, but now I’m not. I see no issue with these injections, and inspected the AST via vim.treesitter.inspect_tree() shows me that the tree-sitter grammar hasn’t changed either. I am at a total loss as to why this isn’t working. Does anyone have any ideas?

**Comment by NikolaIliev** [comment]
Same issue for injecting bash in typescript:

```
(
  (call_expression
    (identifier) @function_name
    (template_string
      (string_fragment) @bash
    )
  )
  (#eq? @function_name "$")
  (#set! @bash "language" "bash")
) @bash!
```

The above does not work and I cannot understand why. Inspecting the AST with :TSPlaygroundToggle shows me everything is correct.

If I remove the eq? predicate, it still does not work.

Interestingly, if I then remove the (identifier), it does start to work. Obviously this isn't what I want though as it targets any template string.

This works, but it's not what I want:

```
(
  (call_expression
    (template_string
      (string_fragment) @bash
    )
  )
  (#set! @bash "language" "bash")
) @bash!
```

Here is the structure reported by :TSPlaygroundToggle

```
                call_expression [107, 12] - [107, 23]
                  function: identifier [107, 12] - [107, 13]
                  arguments: template_string [107, 13] - [107, 23]
                    string_fragment [107, 14] - [107, 22]
```

The above is for the code:
```ts
$`echo foo`
```
